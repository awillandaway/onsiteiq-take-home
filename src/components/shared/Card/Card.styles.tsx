import { onsiteIqPurple } from 'constants/colors';
import styled from 'styled-components';

interface CardWrapperProps {
  backgroundColor?: string;
}

export const CardWrapper = styled.div<CardWrapperProps>`
  border-radius: 4px;
  border: 2px solid ${onsiteIqPurple};
  background-color: ${(props) => (props.backgroundColor ? props.backgroundColor : '#111111')};
  padding: 16px;
  width: 550px;
  display: flex;
  flex-direction: column;
  flex: 0 0 550px;

  @media (max-width: 1400px) {
    width: 400px;
    flex: 0 0 400px;
    font-size: 14px;
  }

  @media (max-width: 500px) {
    width: auto;
    flex: 1;
  }
`;
