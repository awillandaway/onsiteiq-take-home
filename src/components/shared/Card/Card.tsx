import type { ReactNode } from 'react';
import { CardWrapper } from './Card.styles';

interface CardProps {
  children?: ReactNode;
}

export const Card = ({ children }: CardProps) => <CardWrapper>{children}</CardWrapper>;
