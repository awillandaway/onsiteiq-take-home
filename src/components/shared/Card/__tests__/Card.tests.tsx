import renderer from 'react-test-renderer';
import { Card } from '../Card';

describe('Card', () => {
  it('Renders correctly', () => {
    const card = renderer.create(<Card>Test content</Card>).toJSON();
    expect(card).toMatchSnapshot();
  });
});
