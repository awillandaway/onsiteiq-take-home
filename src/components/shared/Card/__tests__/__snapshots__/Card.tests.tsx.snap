// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders correctly 1`] = `
<div
  className="sc-bcXHqe ktOrDV"
>
  Test content
</div>
`;
